### PR TITLE
fix ASAN error in sdl_impl.cpp

### PR DESCRIPTION
**Pull request by oltolm**
This is a fix for an ASAN error.

```
=================================================================
==17172==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x1289a40417e0 at pc 0x7ffaca352bb4 bp 0x00efba4f4710 sp 0x00efba4f4758
READ of size 32 at 0x1289a40417e0 thread T0
    #0 0x7ffaca352bb3 in __asan_memcpy (C:\msys64\clang64\bin\libclang_rt.asan_dynamic-x86_64.dll+0x180042bb3)
    #1 0x7ff6ab2a8ebb in InputCommon::SDL::SDLState::GetSDLControllerButtonBindByGUID(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, 
int, Settings::NativeButton::Values) C:/src/citra/src/input_common/sdl/sdl_impl.cpp:433:17
    #2 0x7ff6a9c2d908 in InputCommon::GetControllerButtonBinds(Common::ParamPackage const&, int) C:/src/citra/src/input_common/main.cpp:107:57
    #3 0x7ff6a8da561a in ConfigureInput::MapFromButton(Common::ParamPackage const&) C:/src/citra/src/citra_qt/configuration/configure_input.cpp:539:21
    #4 0x7ff6a8dae44a in ConfigureInput::AutoMap()::$_0::operator()(Common::ParamPackage const&) const C:/src/citra/src/citra_qt/configuration/configure_input.cpp:566:9
    #5 0x7ff6a8dae3dc in decltype(std::declval<ConfigureInput::AutoMap()::$_0&>()(std::declval<Common::ParamPackage const&>())) std::__1::__invoke[abi:ue170006]<ConfigureInput::AutoMap()::$_0&, Common::ParamPackage const&>(ConfigureInput::AutoMap()::$_0&, Common::ParamPackage const&) C:/msys64/clang64/include/c++/v1/__type_traits/invoke.h:340:25
    #6 0x7ff6a8dae38c in void std::__1::__invoke_void_return_wrapper<void, true>::__call[abi:ue170006]<ConfigureInput::AutoMap()::$_0&, Common::ParamPackage const&>(ConfigureInput::AutoMap()::$_0&, Common::ParamPackage const&) C:/msys64/clang64/include/c++/v1/__type_traits/invoke.h:415:5
    #7 0x7ff6a8dae364 in std::__1::__function::__alloc_func<ConfigureInput::AutoMap()::$_0, std::__1::allocator<ConfigureInput::AutoMap()::$_0>, void (Common::ParamPackage const&)>::operator()[abi:ue170006](Common::ParamPackage const&) C:/msys64/clang64/include/c++/v1/__functional/function.h:192:16
    #8 0x7ff6a8da9690 in std::__1::__function::__func<ConfigureInput::AutoMap()::$_0, std::__1::allocator<ConfigureInput::AutoMap()::$_0>, void (Common::ParamPackage const&)>::operator()(Common::ParamPackage const&) C:/msys64/clang64/include/c++/v1/__functional/function.h:363:12
    #9 0x7ff6a8e07519 in std::__1::__function::__value_func<void (Common::ParamPackage const&)>::operator()[abi:ue170006](Common::ParamPackage const&) const C:/msys64/clang64/include/c++/v1/__functional/function.h:517:16
    #10 0x7ff6a8def7dc in std::__1::function<void (Common::ParamPackage const&)>::operator()(Common::ParamPackage const&) const C:/msys64/clang64/include/c++/v1/__functional/function.h:1168:12
    #11 0x7ff6a8da6a00 in ConfigureInput::SetPollingResult(Common::ParamPackage const&, bool) C:/src/citra/src/citra_qt/configuration/configure_input.cpp:613:9
    #12 0x7ff6a8dd4c51 in ConfigureInput::ConfigureInput(QWidget*)::$_10::operator()() const C:/src/citra/src/citra_qt/configuration/configure_input.cpp:391:17
    #13 0x7ff6a8dd4772 in QtPrivate::FunctorCall<QtPrivate::IndexesList<>, QtPrivate::List<>, void, ConfigureInput::ConfigureInput(QWidget*)::$_10>::call(ConfigureInput::ConfigureInput(QWidget*)::$_10&, void**) C:/msys64/clang64/include/qt6/QtCore/qobjectdefs_impl.h:137:13
    #14 0x7ff6a8dd4661 in void QtPrivate::Functor<ConfigureInput::ConfigureInput(QWidget*)::$_10, 0>::call<QtPrivate::List<>, void>(ConfigureInput::ConfigureInput(QWidget*)::$_10&, void*, void**) C:/msys64/clang64/include/qt6/QtCore/qobjectdefs_impl.h:339:13
    #15 0x7ff6a8dd45fd in QtPrivate::QCallableObject<ConfigureInput::ConfigureInput(QWidget*)::$_10, QtPrivate::List<>, void>::impl(int, QtPrivate::QSlotObjectBase*, QObject*, void**, bool*) C:/msys64/clang64/include/qt6/QtCore/qobjectdefs_impl.h:522:21
    #16 0x7ffacb5b0ffc in void doActivate<false>(QObject*, int, void**) (C:\msys64\clang64\bin\Qt6Core.dll+0x1800d0ffc)
    #17 0x7ffacb5c13a8 in QTimer::timerEvent(QTimerEvent*) (C:\msys64\clang64\bin\Qt6Core.dll+0x1800e13a8)
    #18 0x7ffacb5a5bdf in QObject::event(QEvent*) (C:\msys64\clang64\bin\Qt6Core.dll+0x1800c5bdf)
    #19 0x7ffaceca7a4f in QApplicationPrivate::notify_helper(QObject*, QEvent*) (C:\msys64\clang64\bin\Qt6Widgets.dll+0x180007a4f)
    #20 0x7ffaceca8894 in QApplication::notify(QObject*, QEvent*) (C:\msys64\clang64\bin\Qt6Widgets.dll+0x180008894)
    #21 0x7ffacb562b57 in QCoreApplication::notifyInternal2(QObject*, QEvent*) (C:\msys64\clang64\bin\Qt6Core.dll+0x180082b57)
    #22 0x7ffacb6de259 in QEventDispatcherWin32Private::sendTimerEvent(int) (C:\msys64\clang64\bin\Qt6Core.dll+0x1801fe259)
    #23 0x7ffacb6ddd94 in qt_internal_proc(HWND__*, unsigned int, unsigned long long, long long) (C:\msys64\clang64\bin\Qt6Core.dll+0x1801fdd94)
    #24 0x7ffb04f0e857  (C:\WINDOWS\System32\USER32.dll+0x18000e857)
    #25 0x7ffb04f0e298  (C:\WINDOWS\System32\USER32.dll+0x18000e298)
    #26 0x7ffacb6def39 in QEventDispatcherWin32::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) (C:\msys64\clang64\bin\Qt6Core.dll+0x1801fef39)
    #27 0x7ffacb164174 in QWindowsGuiEventDispatcher::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) (C:\msys64\clang64\bin\Qt6Gui.dll+0x1803c4174)
    #28 0x7ffacb56be24 in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) (C:\msys64\clang64\bin\Qt6Core.dll+0x18008be24)
    #29 0x7ffaceed7ea2 in QDialog::exec() (C:\msys64\clang64\bin\Qt6Widgets.dll+0x180237ea2)
    #30 0x7ff6a919a1f4 in GMainWindow::OnConfigure() C:/src/citra/src/citra_qt/main.cpp:2185:35
    #31 0x7ff6a8bb359c in QtPrivate::FunctorCall<QtPrivate::IndexesList<>, QtPrivate::List<>, void, void (GMainWindow::*)()>::call(void (GMainWindow::*)(), GMainWindow*, void**) C:/msys64/clang64/include/qt6/QtCore/qobjectdefs_impl.h:145:13
    #32 0x7ff6a8bb3329 in void QtPrivate::FunctionPointer<void (GMainWindow::*)()>::call<QtPrivate::List<>, void>(void (GMainWindow::*)(), GMainWindow*, void**) C:/msys64/clang64/include/qt6/QtCore/qobjectdefs_impl.h:182:13
    #33 0x7ff6a8bb2fa7 in QtPrivate::QCallableObject<void (GMainWindow::*)(), QtPrivate::List<>, void>::impl(int, QtPrivate::QSlotObjectBase*, QObject*, void**, bool*) C:/msys64/clang64/include/qt6/QtCore/qobjectdefs_impl.h:520:21
    #34 0x7ffacb5b0ffc in void doActivate<false>(QObject*, int, void**) (C:\msys64\clang64\bin\Qt6Core.dll+0x1800d0ffc)
    #35 0x7ffacb1e8d17 in QAction::activate(QAction::ActionEvent) (C:\msys64\clang64\bin\Qt6Gui.dll+0x180448d17)
    #36 0x7ffacee5418a in QMenuPrivate::activateCausedStack(QList<QPointer<QWidget>> const&, QAction*, QAction::ActionEvent, bool) (C:\msys64\clang64\bin\Qt6Widgets.dll+0x1801b418a)
    #37 0x7ffacee51c22 in QMenuPrivate::activateAction(QAction*, QAction::ActionEvent, bool) (C:\msys64\clang64\bin\Qt6Widgets.dll+0x1801b1c22)
    #38 0x7ffacee58610 in QMenu::mouseReleaseEvent(QMouseEvent*) (C:\msys64\clang64\bin\Qt6Widgets.dll+0x1801b8610)
    #39 0x7ffacecf552d in QWidget::event(QEvent*) (C:\msys64\clang64\bin\Qt6Widgets.dll+0x18005552d)
    #40 0x7ffacee58cb3 in QMenu::event(QEvent*) (C:\msys64\clang64\bin\Qt6Widgets.dll+0x1801b8cb3)
    #41 0x7ffaceca7a4f in QApplicationPrivate::notify_helper(QObject*, QEvent*) (C:\msys64\clang64\bin\Qt6Widgets.dll+0x180007a4f)
    #42 0x7ffaceca9f5e in QApplication::notify(QObject*, QEvent*) (C:\msys64\clang64\bin\Qt6Widgets.dll+0x180009f5e)
    #43 0x7ffacb562b57 in QCoreApplication::notifyInternal2(QObject*, QEvent*) (C:\msys64\clang64\bin\Qt6Core.dll+0x180082b57)
    #44 0x7ffaceca800c in QApplicationPrivate::sendMouseEvent(QWidget*, QMouseEvent*, QWidget*, QWidget*, QWidget**, QPointer<QWidget>&, bool, bool) (C:\msys64\clang64\bin\Qt6Widgets.dll+0x18000800c)
    #45 0x7ffaced0b4af in QWidgetWindow::handleMouseEvent(QMouseEvent*) (C:\msys64\clang64\bin\Qt6Widgets.dll+0x18006b4af)
    #46 0x7ffaced0a1b5 in QWidgetWindow::event(QEvent*) (C:\msys64\clang64\bin\Qt6Widgets.dll+0x18006a1b5)
    #47 0x7ffaceca7a4f in QApplicationPrivate::notify_helper(QObject*, QEvent*) (C:\msys64\clang64\bin\Qt6Widgets.dll+0x180007a4f)
    #48 0x7ffaceca8894 in QApplication::notify(QObject*, QEvent*) (C:\msys64\clang64\bin\Qt6Widgets.dll+0x180008894)
    #49 0x7ffacb562b57 in QCoreApplication::notifyInternal2(QObject*, QEvent*) (C:\msys64\clang64\bin\Qt6Core.dll+0x180082b57)
    #50 0x7ffacae1b588 in QGuiApplicationPrivate::processMouseEvent(QWindowSystemInterfacePrivate::MouseEvent*) (C:\msys64\clang64\bin\Qt6Gui.dll+0x18007b588)
    #51 0x7ffacae79b69 in QWindowSystemInterface::sendWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>) (C:\msys64\clang64\bin\Qt6Gui.dll+0x1800d9b69)
    #52 0x7ffacb6de9cd in QEventDispatcherWin32::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) (C:\msys64\clang64\bin\Qt6Core.dll+0x1801fe9cd)
    #53 0x7ffacb164174 in QWindowsGuiEventDispatcher::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) (C:\msys64\clang64\bin\Qt6Gui.dll+0x1803c4174)
    #54 0x7ffacb56be24 in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) (C:\msys64\clang64\bin\Qt6Core.dll+0x18008be24)
    #55 0x7ffacb56317f in QCoreApplication::exec() (C:\msys64\clang64\bin\Qt6Core.dll+0x18008317f)
    #56 0x7ff6a91b2ac7 in main C:/src/citra/src/citra_qt/main.cpp:3231:18
    #57 0x7ff6a8a91314 in __tmainCRTStartup C:/M/B/src/mingw-w64/mingw-w64-crt/crt/crtexe.c:267:15
    #58 0x7ff6a8a91365 in .l_start C:/M/B/src/mingw-w64/mingw-w64-crt/crt/crtexe.c:188:9
    #59 0x7ffb04a47343  (C:\WINDOWS\System32\KERNEL32.DLL+0x180017343)
    #60 0x7ffb058826b0  (C:\WINDOWS\SYSTEM32\ntdll.dll+0x1800526b0)

0x1289a40417e0 is located 32 bytes before 672-byte region [0x1289a4041800,0x1289a4041aa0)
allocated by thread T0 here:
    #0 0x7ffaca354296 in realloc (C:\msys64\clang64\bin\libclang_rt.asan_dynamic-x86_64.dll+0x180044296)
    #1 0x7ff6ac580f1c in real_realloc C:/src/citra/externals/sdl2/SDL/src/stdlib/SDL_malloc.c:5320:63
    #2 0x7ff6ac581221 in SDL_realloc_REAL C:/src/citra/externals/sdl2/SDL/src/stdlib/SDL_malloc.c:5448:11
    #3 0x7ff6ac54c6dc in SDL_PrivateGameControllerParseElement C:/src/citra/externals/sdl2/SDL/src/joystick/SDL_gamecontroller.c:1019:66
    #4 0x7ff6ac54c1a3 in SDL_PrivateGameControllerParseControllerConfigString C:/src/citra/externals/sdl2/SDL/src/joystick/SDL_gamecontroller.c:1055:13
    #5 0x7ff6ac547b20 in SDL_PrivateLoadButtonMapping C:/src/citra/externals/sdl2/SDL/src/joystick/SDL_gamecontroller.c:1099:5
    #6 0x7ff6ac5479fe in SDL_GameControllerOpen_REAL C:/src/citra/externals/sdl2/SDL/src/joystick/SDL_gamecontroller.c:2147:5
    #7 0x7ff6ab4c63e4 in SDL_GameControllerOpen C:/src/citra/externals/sdl2/SDL/src/dynapi/SDL_dynapi_procs.h:174:1
    #8 0x7ff6ab2abafa in InputCommon::SDL::SDLState::InitJoystick(int) C:/src/citra/src/input_common/sdl/sdl_impl.cpp:499:30
    #9 0x7ff6ab2af8b2 in InputCommon::SDL::SDLState::SDLState() C:/src/citra/src/input_common/sdl/sdl_impl.cpp:866:9
    #10 0x7ff6ab29a2ef in std::__1::__unique_if<InputCommon::SDL::SDLState>::__unique_single std::__1::make_unique[abi:ue170006]<InputCommon::SDL::SDLState>() C:/msys64/clang64/include/c++/v1/__memory/unique_ptr.h:689:30
    #11 0x7ff6ab29a1f9 in InputCommon::SDL::Init() C:/src/citra/src/input_common/sdl/sdl.cpp:14:12
    #12 0x7ff6a9c2b480 in InputCommon::Init() C:/src/citra/src/input_common/main.cpp:50:11
    #13 0x7ff6a9165973 in GMainWindow::InitializeWidgets() C:/src/citra/src/citra_qt/main.cpp:384:5
    #14 0x7ff6a915f4e5 in GMainWindow::GMainWindow(Core::System&) C:/src/citra/src/citra_qt/main.cpp:217:5
    #15 0x7ff6a91b2838 in main C:/src/citra/src/citra_qt/main.cpp:3213:17
    #16 0x7ff6a8a91314 in __tmainCRTStartup C:/M/B/src/mingw-w64/mingw-w64-crt/crt/crtexe.c:267:15
    #17 0x7ff6a8a91365 in .l_start C:/M/B/src/mingw-w64/mingw-w64-crt/crt/crtexe.c:188:9
    #18 0x7ffb04a47343  (C:\WINDOWS\System32\KERNEL32.DLL+0x180017343)
    #19 0x7ffb058826b0  (C:\WINDOWS\SYSTEM32\ntdll.dll+0x1800526b0)

SUMMARY: AddressSanitizer: heap-buffer-overflow (C:\msys64\clang64\bin\libclang_rt.asan_dynamic-x86_64.dll+0x180042bb3) in __asan_memcpy
Shadow bytes around the buggy address:
  0x1289a4041500: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x1289a4041580: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x1289a4041600: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x1289a4041680: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x1289a4041700: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
=>0x1289a4041780: fa fa fa fa fa fa fa fa fa fa fa fa[fa]fa fa fa
  0x1289a4041800: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x1289a4041880: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x1289a4041900: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x1289a4041980: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x1289a4041a00: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
==17172==ABORTING
```

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/7427)

### CODE REVIEW

File: CMakePresets.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (Steveice10) [inline CMakePresets.json:1]: What is this file for? It seems unrelated to the PR topic.

Comment (oltolm) [reply CMakePresets.json:1]: It's an IDE independent way to configure [CMake](https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html). I can remove it.

Comment (Steveice10) [reply CMakePresets.json:1]: I would recommend leaving it out and focusing this PR on just the fix.

**Comment by Steveice10** [comment]
Needs fixes for clang-format errors but otherwise looks good to me.

**Review by FearlessTobi — APPROVED** [review]
